perf/src/deduper.rs

```diff
@@ -85,14 +83,14 @@ impl Deduper {
 
     pub fn dedup_packets_and_count_discards(
         &self,
-        batches: &mut [PacketBatch],
-        mut process_received_packet: impl FnMut(&mut Packet, bool, bool),
+        batches: &mut Vec<TxPacketBatch>,
```

Comment:
A change from `&mut [T]` to `&mut Vec[T]` is reducing generalization.
As the rest of the changes in this patch do not need that, maybe keep `&mut [T]`?